### DISCUSSION TITLE
sync history of all devices with one click?

### DISCUSSION THREAD

**Discussion by h3rm** [emericg/WatchFlower#49] [Ideas]
hi! is it possible to add this function (linux version)? :) maybe with a delay so that not all 15 devices sync at the same time?

and question1: 
[Linux app:]sync history very often fails on flower care's (all manufacturers like vegtrug, CHYBFU, WANFEI etc with newest firmware 3.3.1) and thermobeacon (sensor blue). a few days later syncing the history works without problems. For the Thermabeacon i think it sometimes work to use the sensor blue app on android and sync the logs with the phone and afterwards sync with the watchflower app again.. but i am unsure if it's just luck or really makes a difference...? :)
[Android app:] syncing history works about 3 of 5 times. Trying to sync a few times most of the time works! So the function seems to be very much more stable than in linux.

question2:
the thermobeacon sensors often show about 51 degrees (maybe double to the real temperature?) even if the temperature is about 25 degrees (syncing with the sensor blue android app a few seconds later to double check).. Syncing 2-3 times in watchflower fixes the wrong value and shows the real temperature again. Eg.:
https://i.imgur.com/NEjNM8y.png

is it maybe a bug or just the cheap sensors? Or maybe a problem with a weak battery (with a new one i think there are no problems in the first weeks)?

thx!

**Comment by emericg** [comment]
Hi!

Yes it's possible to add a 'sync all history' button, and queuing these actions so they don't happen all at once. I think two simultaneous might work but that's probably about it. I'll work on that for the next release!

For your first question, I have a lot of sensors (it doesn't help, the place is crowded with bluetooth) and I'm seeing similar symptoms. These device are just not professional grade sensors with perfect reliability...
And you mention using different platforms, which does have a big impact too, because different devices / platforms can have very different bluetooth 'quality' or reliability.
> For the Thermabeacon i think it sometimes work to use the sensor blue app on android and sync the logs with the phone and afterwards sync with the watchflower app again.. but i am unsure if it's just luck or really makes a difference...? :)

definitely luck ^^

Regarding incorrect values with your second thermobeacon, it is weird indeed.
Is it possible that the sensor is configured to use Fahrenheit and the app isn't? Is the humidity value correct (while the temp is wrong) if you compare it to your other thermobeacon?
I never saw this happening on mines, but I can check my code to see if there is any funny business happening.

**Reply by h3rm** [reply]
> > Yes it's possible to add a 'sync all history' button, and queuing these actions so they don't happen all at once. I think two simultaneous might work but that's probably about it. I'll work on that for the next release!

yeah, very cool thx!

> > For your first question, I have a lot of sensors (it doesn't help, the place is crowded with bluetooth) and I'm seeing similar symptoms. These device are just not professional grade sensors with perfect reliability... And you mention using different platforms, which does have a big impact too, because different devices / platforms can have very different bluetooth 'quality' or reliability.

ok, no problem... it's not a very (!) important function, but when the linux-machine hangs up and watchflower doesn't sync a few days it's cool (but i will use android more often ;) )

> > For the Thermabeacon i think it sometimes work to use the sensor blue app on android and sync the logs with the phone and afterwards sync with the watchflower app again.. but i am unsure if it's just luck or really makes a difference...? :) definitely luck ^^

:)

> > Regarding incorrect values with your second thermobeacon, it is weird indeed. Is it possible that the sensor is configured to use Fahrenheit and the app isn't? Is the humidity value correct (while the temp is wrong) if you compare it to your other thermobeacon? I never saw this happening on mines, but I can check my code to see if there is any funny business happening.
the humidtywas correct, only the temperature is high and the calculation of "feels like" is strange... I checked sensorblue app again and did a reset on all three sensors (celsius was configured on all three devices)... in the last days i couldn't reproduce the problem so i think the problem could be a too high temperature for the Thermobeacons... i will keep you up to date it happens again now it's not that hot anymore


Another thing i just recognized when i looked at my screenshot in the first post [Linux]:
In the overview of the mi flora sensors on 25.9. there was the "start date" 12.9. (my first post in this thread: https://i.imgur.com/NEjNM8y.png) - today after sync the "start date" is the 19.9. (https://i.imgur.com/HDztSQ5.png)
-in the monthly view i can see that the logging should be at least since 4.9. (https://i.imgur.com/cJ1phhK.png)

I tink that this didn't happen in the windows app? On Android i see the same dates like in linux.

It would be really cool if it would be possible to extend the logging in the overview and the "calendar-view".. or is it maybe my fault because i synced the android "flowercare"-app on 25.9 and this messed up the history in watchflower somehow? I didn't cleared the history in flowercare and/or in watchflower.. i only played around with syncing the history in watchflower on linux and android and in android flowercare

@emericg i just sent you a little gift via paypal for your quick help and answering (and h3rm-help ;) ) all the time

**Reply by emericg** [reply]
> Another thing i just recognized when i looked at my screenshot in the first post [Linux]:
In the overview of the mi flora sensors on 25.9. there was the "start date" 12.9. (my first post in this thread: https://i.imgur.com/NEjNM8y.png) - today after sync the "start date" is the 19.9. (https://i.imgur.com/HDztSQ5.png)
-in the monthly view i can see that the logging should be at least since 4.9. (https://i.imgur.com/cJ1phhK.png)

It's working as intended I think, the histograms have a 30 days window while the line graphs only have 14 days.

> It would be really cool if it would be possible to extend the logging in the overview and the "calendar-view"

Yes I think I'll add a way to switch between previous/next months/weeks/days, I just have to figure out a clean UI for this.
Also I'll probably add a way to switch between 14/30 days windows for the line graph (at least on desktop).

> @emericg i just sent you a little gift via paypal for your quick help and answering (and h3rm-help ;) ) all the time

Thank you that is much appreciated!
I'm not working very hard on WatchFlower right now, but this is all set in the roadmap for the next release.

**Comment by emericg** [comment]
> question2: the thermobeacon sensors often show about 51 degrees (maybe double to the real temperature?) even if the temperature is about 25 degrees (syncing with the sensor blue android app a few seconds later to double check).. Syncing 2-3 times in watchflower fixes the wrong value and shows the real temperature again. Eg.: https://i.imgur.com/NEjNM8y.png

By the way, does the sensor blue app showed the wrong value too?
I'm adding some code to double check many of the values, but at this time I'm not seeing anything that would explain the temp*2 value. The "feels like 100°" staying displayed even when temperature was back to normal was a bug though.

**Reply by h3rm** [reply]
> By the way, does the sensor blue app showed the wrong value too?

hi! the sensorblue-app didn't show the wrong value in summer, it was "correct" in the app all the time on every sync... but it's really no problem - you don't have to search for this problem.. i think i will change the sensors next year :)

>I'm adding some code to double check many of the values, but at this time I'm not seeing anything that would explain the temp*2 value. >The "feels like 100°" staying displayed even when temperature was back to normal was a bug though.

cool, thx! 👍

**Reply by emericg** [reply]
There is an Android beta out there if you want to give it a go. It has the 'sync all' button.

**Reply by h3rm** [reply]
> 
> 
> There is an Android beta out there if you want to give it a go. It has the 'sync all' button.

hi! very nice, thx! it's syncing atm but i will give you an update when it's ready.. the first three flower care's synced successfull! 👍 

just a quick question: what does the setting "limit bluetooth scanning range"? by default it's "off" :)

**Reply by emericg** [reply]
It's to scan only for sensors close to your phone. Some people in dense urban environments have reported scanning sensors from neighbors :)
There is a legend but it appears only when the option is clicked. Maybe that wasn't the best idea ^^

**Reply by h3rm** [reply]
> 
> 
> There is an Android beta out there if you want to give it a go. It has the 'sync all' button.

> 
> 
> > There is an Android beta out there if you want to give it a go. It has the 'sync all' button.
> 
> hi! very nice, thx! it's syncing atm but i will give you an update when it's ready.. the first three flower care's synced successfull! 👍
> 
> just a quick question: what does the setting "limit bluetooth scanning range"? by default it's "off" :)

syncing of ALL (even 2/3 thermobeacon - the third one seems to be broke because humidity is the same all the time -> but this must be the defect sensor...) 15 sensors worked perfect.. after the sync 10 sensors showed the message "offline" but sync worked perfect for all devices! 👍 GREAT!

ok, maybe one problem/bug: it only synced from 28.10. until today in the overview (in the historic from 13.10 until today) .. so not the last month or better: longer (3 months) :) 

-> really great work! even with the small "bug" mentioned before you solved this perfect! big thx! how often does the app allow to "sync all" - i just tried around and wanted to start a second "sync all" but now it does nothing, hihi.. why did i tried that?:
i thought it makes sense to sync a second time when maybe a few sensors didn't sync correctly. but in my usecase no sync failed so it's not necessary today :)

**Reply by emericg** [reply]
You can 'sync all' all you want ^^ Currently you just can't sync all more than one time at once.

The sync will check for each sensor (with history support) if it needs to be synced (if it's history is too old).
Once synced, each sensor set a "last sync" date (that you can see in the _sensor limit_ tab, for flower care at least). Once a sync is restarted it will be restarted from that date.

I believe that in some cases, it can be wrongly updated when a sync fails (or some other weird thing happen). I'll have to see what's wrong with these cases.

> after the sync 10 sensors showed the message "offline" but sync worked perfect for all devices! ￼ GREAT!

Yes I saw that too, a successful "history sync" is not considered like a successful _regular_ update. That should be the case most of the time though, I'll improve that.

> ok, maybe one problem/bug: it only synced from 28.10. until today in the overview (in the historic from 13.10 until today) .. so not the last month or better: longer (3 months) :)

28.09 is exactly two weeks ago, so it's probably not about the data, but just the graph that's showing only the last two weeks. In computer you can have more data, but that's kind of related to your screen size :) If I use the line graph, it does only show data from 28.09.
As I said previously I'll make these graphs scrollable at some point.

![Capture d’écran de 2021-10-12 18-10-35](https://user-images.githubusercontent.com/691258/136992744-4e62c04b-0621-4c1a-84f9-a8760a74a621.png)

**Reply by h3rm** [reply]
hi @emericg ! 

just fyi :-) i just downloaded the linux watchflower 3.0 appimage and "history sync "seems to work (like before) on the thermobeacons and sometimes on one of the flowercare sensors. When i press the button again i can see that the app wants to sync the sensors which are not correctly/fully synced. I know you already told me that there are a lot of factors, but i just wanted to give you an update. 

two upcoming questions :)

1. i thought in an .appimage file there is all userdata integrated, but i just startet 3.0 and all plant names/settings/etc. are here :) Can you maybe tell me where to find these additional files so i can do a backup? sry for this but i am new to linux ;)

2. It would be cool to have a UI-scale function :) Watchflower 3,0 runs here on an 1280pixel display (urgs) old hp notebook and at the moment i only have 3 sensors in a row. In 2.1 there were 4 sensors per row :) If i could choose the scaling: there could also be 6 plants per row.. 
....it's ok if you can save this feature-request for a future version and maybe on a 1080p display it already looks much better (i dont know :) ). I understand that no one codes an app for this resolution any more... ;) 

thx again for your really cool work! i will try to use the android app more often in the future but on a computer i like the handling/mouse and the screen size very much more, so both systems will run watchflower :) 

[update linux-app after 6x "sync all":] 5 sensors are historical synced - 5 sensors can't be synced. i think i will try to delete the local cache of the failing 5 sensors (btw. cool that there is no option in the off. app ;) ) and afterwards doublecheck everything with your android app when i have time for this..
 
i think that maybe the sensor-cache could be "too full" so it's too much data and the sync fails...? some of the sensors have more than 100 days of logging in the off. flower care app.... i am not sure how to find out how many days the onboard-sensor-cache stores the data so maybe i will try to "clear/reset" the failing sensors :) 

please: you don't have to check or code anything - i just wanted to let you know because i want to find out if this could be the reason ;)

**Reply by emericg** [reply]
> i thought in an .appimage file there is all userdata integrated, but i just startet 3.0 and all plant names/settings/etc. are here :) Can you maybe tell me where to find these additional files so i can do a backup? sry for this but i am new to linux ;)

No the user data are in the user folder, just like other apps.
You can find the database in ~/.local/share/WatchFlower/WatchFlower/ and the settings in ~/.config/WatchFlower/WatchFlower/

> It would be cool to have a UI-scale function :) Watchflower 3,0 runs here on an 1280pixel display (urgs) old hp notebook and at the moment i only have 3 sensors in a row. In 2.1 there were 4 sensors per row :) If i could choose the scaling: there could also be 6 plants per row..

I changed the scaling so the widgets are a bit bigger indeed. It's just very difficult to look good at every resolution and dpi combination there is in the wild unfortunately. 3 sensors in a row for a 720p screen is already pretty good though, I'm not sure that 4 was able to fit all texts and icons correctly?

> i think that maybe the sensor-cache could be "too full" so it's too much data and the sync fails...?

WatchFlower keeps data for 90 days. FlowerCares keep data in their memory for 30 days, ThermoBeacons for 90.
Don't worry about having too much data ^^ again with the sync, sometimes it just doesn't work very well.

> i am not sure how to find out how many days the onboard-sensor-cache stores the data so maybe i will try to "clear/reset" the failing sensors :)

I have that info internally but it doesn't make much sense to show it in the UI.
There is the "last sync" date in the "limits" screen for flower cares though. The history will not be synced beyond that date (because it should be synced already :p).

**Reply by h3rm** [reply]
> No the user data are in the user folder, just like other apps. You can find the database in ~/.local/share/WatchFlower/WatchFlower/ and the settings in ~/.config/WatchFlower/WatchFlower/

cool, i just did a backup, thx for the info!

> I changed the scaling so the widgets are a bit bigger indeed. It's just very difficult to look good at every resolution and dpi combination there is in the wild unfortunately. 3 sensors in a row for a 720p screen is already pretty good though, I'm not sure that 4 was able to fit all texts and icons correctly?

I compared both versions and now i like the new scaling more because i can see longer sensor names in the overview :) so scaling the UI is not important any more after a few days of working with it!

FYI if you are interested how it looks like (at the moment only 13/15 sensors are in use):
v2.1 on 1366x768 pixel 15.4" display: https://i.imgur.com/lJ5Ag5A.png
v3.0 on 1366x768 pixel 15.4" display: https://i.imgur.com/qrNpv8i.png
(sorry 1366x768 not 1280x720 and i just found out that i can virtually scale the resolution!)

> WatchFlower keeps data for 90 days. FlowerCares keep data in their memory for 30 days, ThermoBeacons for 90. Don't worry about having too much data ^^ again with the sync, sometimes it just doesn't work very well.

thanks for this information! 👍 deleting the sensor cache didn't made any difference, you are right ;) but still interesting why history sync works nearly every time on some sensors and never work on some other sensors (all the time the same devices ;) )... but using the android app parallel is a good workaround.. also i don't need the history sync often but it's great when the notebook shuts down and i recognize it a few days later ... syncing all 2 hours (configured in the watchflower settings) works perfect if the notebook "works perfect"... ;) 

> I have that info internally but it doesn't make much sense to show it in the UI. There is the "last sync" date in the "limits" screen for flower cares though. The history will not be synced beyond that date (because it should be synced already :p).

You don't have to display this info in the app! :) i was only interested if this could cause the problem. If i understand that correct maybe it's not the best idea to use android and linux app parallel cause android will set the "last sync date" for each sensor-memory or did you mean  within the app(s)?

Thanks again for your reply! After a few days using V3 i am really happy with both linux and android apps! it's a really wonderful app compared to the official one which really sucks and is extremly slow ... ;)

**Reply by h3rm** [reply]
hi! i just had another idea that it would be cool to have a memo/note function (freetext) for each sensor so that i don't have to use an external tool to write down things like last fertilizing, last watering, etc (it's not good to 100% trust especially the EC-values of the flowercares, so i write down everything in memos too). maybe with a mouseover in the main overview so that its possible to quick check all sensors (or maybe in the sensor X/Y/Z overview of each sensor) in linux. maybe this could be difficult to display in the android app but i think it would be a nice feature in the desktop apps :)

**Reply by emericg** [reply]
Sorry I just realized I never answered you.
What you propose is similar to what is described in that issue : https://github.com/emericg/WatchFlower/issues/14
I'll probably do it at some point, it's a nice feature, but not today sorry :p
